std/math.d

```diff
@@ -4268,62 +4268,25 @@ real remquo(real x, real y, out int n) @trusted nothrow @nogc  /// ditto
 struct IeeeFlags
 {
 private:
+    import core.stdc.fenv;
+
     // The x87 FPU status register is 16 bits.
     // The Pentium SSE2 status register is 32 bits.
+    // PowerPC FPSCR is a 32-bit register.
+    // ARM FPSCR is a 32bit register
+    // SPARC FSR is a 32bit register (64 bits for SPARC 7 & 8, but high bits are uninteresting).
     uint flags;
-    version (X86_Any)
-    {
-        // Applies to both x87 status word (16 bits) and SSE2 status word(32 bits).
-        enum : int
-        {
-            INEXACT_MASK   = 0x20,
-            UNDERFLOW_MASK = 0x10,
-            OVERFLOW_MASK  = 0x08,
-            DIVBYZERO_MASK = 0x04,
-            INVALID_MASK   = 0x01
-        }
-        // Don't bother about subnormals, they are not supported on most CPUs.
-        //  SUBNORMAL_MASK = 0x02;
-    }
-    else version (PPC_Any)
-    {
-        // PowerPC FPSCR is a 32-bit register.
-        enum : int
-        {
-            INEXACT_MASK   = 0x02000000,
-            DIVBYZERO_MASK = 0x04000000,
-            UNDERFLOW_MASK = 0x08000000,
-            OVERFLOW_MASK  = 0x10000000,
-            INVALID_MASK   = 0x20000000 // Summary as PowerPC has five types of invalid exceptions.
-        }
-    }
-    else version (ARM)
+
+    enum : int
     {
-        // ARM FPSCR is a 32bit register
-        enum : int
-        {
-            INEXACT_MASK   = 0x10,
-            UNDERFLOW_MASK = 0x08,
-            OVERFLOW_MASK  = 0x04,
-            DIVBYZERO_MASK = 0x02,
-            INVALID_MASK   = 0x01
-        }
+        INEXACT_MASK   = FE_INEXACT,
+        UNDERFLOW_MASK = FE_UNDERFLOW,
+        OVERFLOW_MASK  = FE_OVERFLOW,
+        DIVBYZERO_MASK = FE_DIVBYZERO,
+        INVALID_MASK   = FE_INVALID
```

Comment:
Why do we even need those? Wouldn't it be better to directly replace all occurrences? 
As far as I can tell the enum is private